src/zkevm_specs/evm/execution/oog_call.py

```diff
@@ -30,28 +29,4 @@ def oog_call(instruction: Instruction):
     gas_not_enough, _ = instruction.compare(instruction.curr.gas_left, gas_cost, N_BYTES_GAS)
     instruction.constrain_equal(gas_not_enough, FQ(1))
 
-    # current call must be failed.
-    instruction.constrain_equal(
-        instruction.call_context_lookup(CallContextFieldTag.IsSuccess), FQ(0)
-    )
-
-    # Go to EndTx only when is_root
-    is_to_end_tx = instruction.is_equal(instruction.next.execution_state, ExecutionState.EndTx)
-    instruction.constrain_equal(FQ(instruction.curr.is_root), is_to_end_tx)
-
-    # state transition.
-    if instruction.curr.is_root:
-        # Do step state transition
-        instruction.constrain_step_state_transition(
-            rw_counter=Transition.delta(12),
-            call_id=Transition.same(),
-        )
-    else:
-        # when it is internal call, need to restore caller's state as finishing this call.
-        # Restore caller state to next StepState
-        instruction.step_state_transition_to_restored_context(
-            rw_counter_delta=12,
-            return_data_offset=FQ(0),
-            return_data_length=FQ(0),
-            gas_left=instruction.curr.gas_left,
-        )
+    instruction.constrain_error_state(12)
```

Comment:
It seems originally the `reversible_write_counter` is not added here?

```suggestion
    instruction.constrain_error_state(12 + instruction.curr.reversible_write_counter.n)
```

Reply:
Fixed `rw_counter_delta = 12 + instruction.curr.reversible_write_counter.n` in OOG Call in commit e6b9653.
And updated test cases to set `reversible_write_counter = 0` for root Call, and `reversible_write_counter = 2` (parent Call and Stop) for internal Call.

Reply:
I think the updated test cases are setting caller's `reversible_write_counter`, and here it's actually using callee's `reversible_write_counter`, we should update all the `callee_reversible_write_counter` to non-zero values at some points to make sure the logic here is okay (there are other exception state also using `0`).